modules/OsmOsisManager.py

```diff
@@ -322,57 +284,48 @@ def run_diff(self, conf):
         self.logger.log("iteration=%d" % nb_iter)
 
         try:
-          cmd  = [conf.bin_osmosis]
-          cmd += ["--read-replication-interval", "workingDirectory=%s" % diff_path]
-          cmd += ["--simplify-change", "--write-xml-change", "file=%s" % xml_change]
-          cmd += ["-quiet"]
+          cmd  = [conf.bin_pyosmium_up_to_date]
+          cmd += ["-v"]
+          cmd += ["--format", "pbf"]
+          cmd += ["--outfile", tmp_pbf_file]
+          cmd += ["--tmpdir", diff_path]
+          cmd += ["--force-update-of-old-planet"]
+          cmd += [pbf_file]
           self.logger.execute_err(cmd)
         except:
           self.logger.log("waiting 2 minutes")
           time.sleep(2*60)
           continue
 
-        cmd  = [conf.bin_osmosis]
-        cmd += ["--read-xml-change", "file=%s" % xml_change]
-        cmd += ["--read-pbf", "file=%s" % conf.download["dst"] ]
-        cmd += ["--apply-change", "--buffer"]
-        cmd += ["--write-pbf", "file=%s" % tmp_pbf_file]
-        cmd += ["-quiet"]
-        self.logger.execute_err(cmd)
-
-        shutil.move(tmp_pbf_file, conf.download["dst"])
-
-        # find if state.txt is more recent than one day
-        osm_state = OsmState(os.path.join(diff_path, "state.txt"))
-        cur_ts = datetime.datetime.today()
-        print("state: ", nb_iter, " - ", osm_state.timestamp(), end=' ')
-        if prev_state_ts is not None:
-          print("   ", prev_state_ts - osm_state.timestamp())
-        if osm_state.timestamp() > (cur_ts - datetime.timedelta(days=1)):
+        shutil.move(tmp_pbf_file, pbf_file)
+
+        # Check if state.txt is more recent than one day
+        osmobject = osmium.io.Reader(pbf_file)
+        cur_timestamp = dateutil.parser.isoparse(osmobject.header().get('osmosis_replication_timestamp'))
+        print("state: ", nb_iter, " - ", cur_timestamp, end=' ')
+        print("   ", prev_timestamp - cur_timestamp)
+        if cur_timestamp > (cur_ts - datetime.timedelta(days=1)):
           is_uptodate = True
-        elif prev_state_ts == osm_state.timestamp():
+        elif prev_timestamp == cur_timestamp:
           is_uptodate = True
         else:
-          prev_state_ts = osm_state.timestamp()
+          prev_timestamp = cur_timestamp
 
       if not is_uptodate:
         # we didn't get the latest version of the pbf file
         self.logger.log(self.logger.log_av_r + "didn't get latest version of osm file" + self.logger.log_ap)
         return (False, None)
       elif nb_iter == 1:
-        return (True, xml_change)
+        return (True, None)
```

Comment:
On abandonne toute idée d'utiliser les diffs, du coup ?

Reply:
For now, with osmium.